irods/connection.py

```diff
@@ -107,7 +107,8 @@ def recv(self, into_buffer = None
                 msg = iRODSMessage.recv(self.socket)
             else:
                 msg = iRODSMessage.recv_into(self.socket, into_buffer)
-        except socket.error:
+        except (socket.error, socket.timeout) as e:
```

Comment:
If _recv_message_in_len() fails in recv() or recv_into(), it will now throw a socket.error exception. The exception is caught here now, a critical message is logged to alert the admins, and its wrapped in a NetworkException with a more user friendly message.